workflow/events/event_recorder_manager.go

```diff
@@ -11,6 +11,9 @@ import (
 	"k8s.io/client-go/tools/record"
 )
 
+// by default, allow a source to send 10000 events about an object
+const defaultSpamBurst = 10000
```

Comment:
This will support to trigger WorkflowSucceeded event if the workflow has 10000 steps. This change won't the default EventAggregator config.